### DISCUSSION TITLE
Use GlobalOrderingPoint for SingleCycleUniqueEvent

### DISCUSSION THREAD

**Discussion by CraneXJH** [sparcians/map#499] [General]
Hi, I want to use GlobalOrderingPoint () to scheduler my SingleCycleUniqueEvent. I found it seems the two events should be in the same node. How can I scheduler two events in two different node?

**Comment by klingaard** [comment]
Not sure I follow what you're asking. GlobalOrderingPoints (GOP) are not used for scheduling directly but used to ensure one event is always scheduled before or after another event.  But perhaps you already know that and are asking a different question. 

GOP is documented here:
https://sparcians.github.io/map/classsparta_1_1GlobalOrderingPoint.html#details

To use it, you must supply a node in your simulation tree for it to find registered GOPs.  That node can be any node in the tree.

**Comment by klingaard** [comment]
To be more specific, you don't need to create a GOP in one place and use it in multiple places. Instead, you can "create it" in multiple places and use it immediately there.  The framework is smart enough to avoid the duplication. 

Think of it this way: you're asking the framework to "either create new or find existing" GOP with the given name and use it

**Comment by CraneXJH** [comment]
@klingaard Thank you for your anwser, now I know how to schedule two events. How about schedule three events?
For example, there are three events in one node, their order is event1 >> event2 >>event3. Can I use the following code to schedule them？
`event1 >> GlobalOrderingPoint(node, "schedule_event");`
`GlobalOrderingPoint(node, "schedule_event") >> event2;`
`GlobalOrderingPoint(node, "schedule_event") >>event3;`
Or I also need to set the order between event2 and event3?

**Comment by klingaard** [comment]
Assuming that events 1, 2, and 3 are in different units, you would do this:

Unit1
```
   event1 >> GlobalOrderingPoint(node, "schedule_event1");
```
Unit2
```
   GlobalOrderingPoint(node, "schedule_event1") >> event2 >> GlobalOrderingPoint(node, "schedule_event2");
```
Unit3
```
   GlobalOrderingPoint(node, "schedule_event2") >> event3;
```
If I'm understanding your use case correctly...
